components/isceobj/Sensor/UAVSAR_HDF5_SLC.py

```diff
@@ -247,9 +263,17 @@ def extractImage(self):
         ds = fid['/science/LSAR/SLC/swaths/' + self.frequency + '/' + self.polarization]
         nLines = ds.shape[0]
 
-        with open(self.output, 'wb') as fout:
-            for ii in range(nLines):
-                ds[ii,:].astype(np.complex64).tofile(fout)
+        # if TypeError is raised (e.g. complex32), force casting to complex64
+        try:
+            _ = ds.dtype
+            sink = DummySink()
+        except TypeError:
+            sink = ds.astype(np.complex64)
+
+        with sink:
```

Comment:
Wouldn't the case below always work. 
```
with ds.astype(np.complex64) as sink:
```

Can remove the try/ except etc then?

Reply:
Yes, but if there's no error, e.g. input is complex64, complex128, or even real (although that would be unexpected for SLCs), we don't need to force casting to complex64..

Reply:
True. Unfortunately, I dont think we have mechanisms in place in isce2 to handle SLCs other than complex64.

Reply:
I see. Do you prefer that I force casting to complex64?

Reply:
I think for ISCE2, we have to. Since the XML creation that follows essentially says its a CFLOAT and all the SLC reading uses std::complex<float> without a caster check like we do for DEMs.

Reply:
Ok, will do that. Thanks!